Simplified/NYPLBook.m

```diff
@@ -334,22 +368,16 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary
   self.updated = [NSDate dateWithRFC3339String:dictionary[UpdatedKey]];
   if(!self.updated) return nil;
   
-  NSString *const annotations = NYPLNullToNil(dictionary[AnnotationsURLKey]);
-  self.annotationsURL = annotations ? [NSURL URLWithString:annotations] : nil;
-  
-  NSString *const alternate = NYPLNullToNil(dictionary[AlternateURLKey]);
-  self.alternateURL = alternate ? [NSURL URLWithString:alternate] : nil;
-
-  NSString *const analytics = NYPLNullToNil(dictionary[AnalyticsURLKey]);
-  self.analyticsURL = analytics ? [NSURL URLWithString:analytics] : nil;
-  
-  
   return self;
 }
 
 - (NSDictionary *)dictionaryRepresentation
 {
   return @{AcquisitionKey: [self.acquisition dictionaryRepresentation],
+           AlternateURLKey: NYPLNullFromNil([self.alternateURL absoluteString]),
+           AnnotationsURLKey: NYPLNullFromNil([self.annotationsURL absoluteString]),
+           AnalyticsURLKey: NYPLNullFromNil([self.analyticsURL absoluteString]),
+           AuthorLinksKey: self.authorLinks,
```

Comment:
`self.authorLinks` is an `NSArray` of `NYPLOPDSLink` objects. The purpose of `dictionaryRepresentation` is to return an `NYPLBook` that is JSON-serializable. I'm not sure how this can work since `NYPLOPDSLink` is not one of the Cocoa classes serializable by `NSJSONSerialization`. What am I missing here?